src/util.tcc

```diff
@@ -19,11 +19,10 @@ template<typename T> T swap_bit_endianness(T v)
     return v;
 }
 
-// Takes a binary string and swaps the byte endianness
-template<typename FieldT>
-std::vector<FieldT> swap_byte_endianness(std::vector<FieldT> value)
+// Takes a string and swaps the byte endianness
+template<typename T> T swap_byte_endianness(T v)
```

Comment:
The current implementation works for containers of booleans with a `size` method.
Please add a comment to specify that, since it is very easy for the codex base to become a nightmare to maintain with templatized code taking a type `T`.
Btw, the previous function `std::vector<FieldT> swap_byte_endianness(std::vector<FieldT> value)` was lacking comments to explicit the fact that the vector was a vector of field elements which were supposed to be booleans... The function wouldn't have make sense with arbitrary FieldT

Reply:
Corrected